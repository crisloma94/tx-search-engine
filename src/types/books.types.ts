// Defined in DEFAULT_BOOK_FIELDS
export interface BookSearchModel {
  author_name?: string[];
  cover_i?: number;
  first_publish_year?: number;
  id_goodreads?: string[];
  id_librarything?: string[];
  isbn?: string[];
  key: string;
  oclc?: string[];
  publisher?: string[];
  title: string;
}
