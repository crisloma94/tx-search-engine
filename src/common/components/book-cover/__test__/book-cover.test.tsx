import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BookSearchModel } from '../../../../types/books.types';
import Theme from '../../../../styles/theme';
import { store } from '../../../../core/store/store';
import BookCover from '../book-cover.component';

describe('Book cover component test suite', () => {
  const getComponent = (book: BookSearchModel) => (
    <Provider store={store}>
      <Theme>
        <BookCover book={book} />
      </Theme>
    </Provider>
  );
  const setup = (book: BookSearchModel) => render(getComponent(book));

  it('Should match snapshot', () => {
    const { container } = setup({
      author_name: ['Author name test'],
      cover_i: 5678,
      first_publish_year: 1984,
      id_goodreads: ['1234'],
      id_librarything: ['567'],
      isbn: ['89'],
      key: '/works/OL545572W',
      oclc: ['1', '2'],
      publisher: ['Publisher example'],
      title: 'Title test',
    });
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot with default cover', () => {
    const { container } = setup({
      author_name: ['Author name test'],
      first_publish_year: 1984,
      key: '/works/OL545572W',
      publisher: ['Publisher example'],
      title: 'Title test',
    });
    expect(container).toMatchSnapshot();
  });
});
