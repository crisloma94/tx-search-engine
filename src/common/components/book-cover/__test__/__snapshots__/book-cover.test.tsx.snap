// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book cover component test suite Should match snapshot 1`] = `
<div>
  <img
    alt="Default book cover"
    class="sc-bgqQcB ccLtkH book-cover"
    src="https://covers.openlibrary.org/b/id/5678-M.jpg"
  />
</div>
`;

exports[`Book cover component test suite Should match snapshot with default cover 1`] = `
<div>
  <img
    alt="Default book cover"
    class="sc-bgqQcB ccLtkH book-cover"
    src="https://fakeimg.pl/80x112/9381ff/9381ff"
  />
</div>
`;
